### PR TITLE
Clean up and add docs for BigGAN demo

### PR DESCRIPTION
**Pull request by AzizZayed**
## Description ##

Remove all unnecessary classes from the BigGAN demo and reduced it from 5 to 3 files. I used an array as input to the translator instead of a sample-size parameter. I input the 'truncation' parameter as an argument in the criteria. Thus, the translator only needs to take an integer array of category-IDs.

### CODE REVIEW

**Review by frankfliu — APPROVED** [review]

**Comment by frankfliu** [comment]
Let's put the images in our resources s3 bucket.

File: examples/docs/biggan.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+./gradlew run -Dmain=ai.djl.examples.inference.BigGAN -Dai.djl.default_engine=PyTorch
+```
+
+### Output
+
+Your output will vary since the generation depends on a random seed. Here are a few examples:
+
+Black Swan                 |  Golden Retriever          |  Bubble |  Alp  |  Cheeseburger
+:-------------------------:|:-------------------------: |:-------------------------: | :----------------------: | :----------------------:
+![](https://djl-ai.s3.amazonaws.com/resources/images/biggan/black-swan.png) | ![](https://djl-ai.s3.amazonaws.com/resources/images/biggan/golden-retriever.png)| ![](https://djl-ai.s3.amazonaws.com/resources/images/biggan/bubble.png) | ![](https://djl-ai.s3.amazonaws.com/resources/images/biggan/hills.png) | ![](https://djl-ai.s3.amazonaws.com/resources/images/biggan/cheeseburger.png)
```
Comment (frankfliu) [inline examples/docs/biggan.md:42]: use https://resources.djl.ai/images/... url

File: api/src/main/java/ai/djl/Application.java [hunk 2]
```diff
@@ -181,6 +181,9 @@ public int hashCode() {
          * String}&gt;.
          */
         Application WORD_RECOGNITION = new Application("cv/word_recognition");
+
+        /** An application that accepts a seed and returns generated images. */
+        Application GAN = new Application("cv/gan");
```
Comment (zachgk) [inline api/src/main/java/ai/djl/Application.java:186]: Maybe use Generator instead. A GAN is a technique and an Application is supposed to be a problem

Comment (AzizZayed) [reply api/src/main/java/ai/djl/Application.java:186]: How about "GENERATION" to keep with the standard in Application

**Pull request merged**